waiter/src/waiter/token.clj

```diff
@@ -516,48 +516,61 @@
                            :log-level :warn}))))
 
       nil
-      (do
+      (let [existing-editor (get existing-token-metadata "editor")
+            existing-owner (get existing-token-metadata "owner")
```

Comment:
We used to have code that allowed for tokens to not have an existing owner.  Does this break that backward compatibility?

Reply:
What if a token doesn't have a `run-as-user` or `run-as-user` is `*` or ...

Reply:
Oops!

I changed the code to account for a missing owner. This should not break backward compatibility now.

> A weird situation can happen when setting an editor on a token that doesn't have an owner. e.g. If run-as-user is set to me, and owner is null, and I set editor to me as well, I can no longer change the owner, editor, or run-as-user.
> https://github.com/twosigma/waiter/pull/1313#discussion_r602824305

Newly created/modified tokens always have an owner. A token with an editor, will not have a `null` owner.